packages/adp-tooling/src/prompts/change-data-source/index.ts

```diff
@@ -3,7 +3,45 @@ import type { ManifestNamespace } from '@sap-ux/project-access';
 import type { ChangeDataSourceAnswers } from '../../types';
 import { t } from '../../i18n';
 import { filterDataSourcesByType } from '@sap-ux/project-access';
-import { validateEmptyString } from '@sap-ux/project-input-validator';
+import { validateEmptyString, isDataSourceURI } from '@sap-ux/project-input-validator';
+
+/**
+ * Validates the OData Source URI prompt.
+ *
+ * @param value The value to validate.
+ * @returns {boolean | string} True if the value is a valid URI, or an error message if not a valid URI or empty.
+ */
+function validatePromptOdataURI(value: string): boolean | string {
+    const validationResult = validateEmptyString(value);
+    if (typeof validationResult === 'string') {
+        return validationResult;
+    }
+
+    if (!isDataSourceURI(value)) {
+        return t('validators.errorInvalidDataSourceURI');
+    }
+
+    return true;
+}
+
+/**
+ * Validates the Annotation URI prompt.
+ *
+ * @param value The value to validate.
+ * @returns {boolean | string} True if the value is a valid URI or empty, or an error message if not a valid URI.
+ */
+function validatePromptAnnotationURI(value: string): boolean | string {
```

Comment:
Would it make sense to try to join these two methods as the only difference is if `validationResult` is a string or a boolean?

Reply:
I see your point, but I didn't see a nice way to merge to the functions and not make the code uglier. I would either have to pass a parameter to know for which prompt the new functions is executing or have some logic inline in the prompt to parse the result. In both cases the I wouldn't have the nice looking passing of the validator functions directly in the prompts as it currently is. If you have an idea that I don't currently see, please suggest :)

Reply:
Happy with explanation to cover the existing implementation. Thanks.